src/metabase/query_processor/streaming/xlsx.clj

```diff
@@ -1,17 +1,225 @@
+        (get-in currency/currency [(keyword currency-code) :symbol])
+        ;; Fall back to using code if symbol isn't not supported on the Metabase frontend
+        currency-code)
+
+      "code"
+      currency-code
+
+      "name"
+      (get-in currency/currency [(keyword currency-code) :name_plural]))))
+
+(defn- currency-format-string
+  "Adds a currency to the base format string as either a suffix (for pluralized names) or
+  prefix (for symbols or codes)."
+  [base-string format-settings]
+  (let [currency-code (::mb.viz/currency format-settings "USD")
+        currency-identifier (currency-identifier format-settings)]
+    (condp = (::mb.viz/currency-style format-settings "symbol")
+      "symbol"
+      (if (currency/supports-symbol? currency-code)
+        (str "[$" currency-identifier "]" base-string)
+        (str "[$" currency-identifier "] " base-string))
+
+      "code"
+      (str "[$" currency-identifier "] " base-string)
+
+      "name"
+      (str base-string "\" " currency-identifier "\""))))
+
+(defn- default-number-format?
+  "Use default formatting for decimal number types that have no other format settings defined
+  aside from prefix, suffix or scale."
+  [format-settings]
+  (and
+   ;; This is a decimal number (not a currency, percentage or scientific notation)
+   (or (= (::mb.viz/number-style format-settings) "decimal")
+       (not (::mb.viz/number-style format-settings)))
+   ;; Custom number formatting options are not set
+   (not (seq (dissoc format-settings
+                     ::mb.viz/number-style
+                     ::mb.viz/scale
+                     ::mb.viz/prefix
+                     ::mb.viz/suffix)))))
+
+(defn- number-format-strings
+  "Returns format strings for a number column corresponding to the given settings.
+  The first value in the returned list should be used for integers, or numbers that round to integers.
+  The second number should be used for all other values."
+  [format-settings semantic-type]
+  (let [format-strings
+        (let [decimals        (::mb.viz/decimals format-settings 2)
+              is-currency?    (or (isa? semantic-type :type/Currency)
+                                  (= (::mb.viz/number-style format-settings) "currency"))
+              merged-settings (if is-currency?
+                                (merge-global-settings format-settings :type/Currency)
+                                format-settings)
+              base-strings    (if (default-number-format? merged-settings)
+                                ;; [int-format, float-format]
+                                ["#,##0", "#,##0.##"]
+                                (repeat 2 (apply str "#,##0" (when (> decimals 0) (apply str "." (repeat decimals "0"))))))]
+          (condp = (::mb.viz/number-style merged-settings)
+            "percent"
+            (map #(str % "%") base-strings)
+
+            "scientific"
+            (map #(str % "E+0") base-strings)
```

Comment:
this isn't actual scientific notation but I bet we punted on this somewhere. might add a ticket for figuring that out one day

Reply:
do you mean "scientific notation" != "E notation"?

on the frontend we display stuff in scientific notation, but this is the closest we can get in Excel